cfme/cloud/keypairs.py

```diff
@@ -35,6 +35,11 @@ def __init__(self, name, provider, appliance=None):
         self.name = name
         self.provider = provider
 
+    def load_details(self, refresh=False):
```

Comment:
I think you should put the refresh into the Details destination's resetter(), and get rid of this function. I don't see an advantage over using navigate_to(obj, 'Details').